Filter/StringFilter.php

```diff
@@ -55,7 +55,10 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
         if ($data['type'] == ChoiceType::TYPE_EQUAL) {
             $queryBuilder->setParameter($parameterName, $data['value']);
         } else {
-            $queryBuilder->setParameter($parameterName, sprintf($this->getOption('format'), $data['value']));
+            $queryBuilder->setParameter($parameterName, sprintf(
+                $this->getOption('format'),
+                addcslashes($data['value'], '%_')
```

Comment:
What if someone currently uses '%' and '_' chars as a feature for searching? It's not BC.
Also, '\\%' is escaped to '\\\\%', which evaluates as *"search a backslash followed by anything else"*. The escaping strategy is wrong.